src/App.tsx

```diff
@@ -1,61 +1,138 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId)
+    || null;
+}
+
+function getMaxTodoId(todos: Todo[]) {
+  const todoIds = todos.map(({ id }) => id);
+
+  return Math.max(...todoIds);
+}
+
+export const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const [todos, setTodos] = useState<Todo[]>(preparedTodos);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (!title && !userId) {
+      setHasUserIdError(true);
+      setHasTitleError(true);
+
+      return;
+    }
+
+    if (!title) {
+      setHasTitleError(true);
+
+      return;
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+
+      return;
+    }
```

Comment:
Try to write this condition a little shorter:
```suggestion
    if (!title || !userId) {
      if (!title) {
        setHasTitleError(true);
      }
     
      if (!userId) {
        setHasUserIdError(true);
      }

      return;
    }
```